# Business Understanding
## Overview of the business
A company that is seeking to expand into the aviation industry and wants buy aircrafts for its new division

## Business Problem

Being new in the aviation industry the company is unaware of the potential risks that are associated with aircrafts 

## Business Objectives
The company wants to understand the potential risks it may face in the aviation industry
The company wants to obtain a list of recommended aircrafts that are considered to have low risks in regards to its operational use

## Success Criteria
To obtain a list of aircrafts that are considered to have low risks

To outline the potential risks the company may experience in the aviation industry

To produce relevant visualisations that provide insights on aircraft and the aviation industry

## Project plan

To go through a dataset that contains accidents that have occurred in the aviation industry and to obtain relevant insights that can be able to solve the business problem






# Familization with the dataset

In [2]:
# Importing the pandas and matplotlib libraries
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

In [3]:
pd.options.display.max_columns = None

Loading the dataset to be used


In [4]:
aviation_data_df = pd.read_csv("data/AviationData.csv", encoding='latin-1', dtype='str')


In [5]:
usstate_codes_df = pd.read_csv("data/USState_Codes.csv")

Checking the head, tail and a sample of the dataset so to get an understanding of the dataset we are working with


In [6]:
aviation_data_df.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude   Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2    Saltville, VA  United States  36.922223  -81.878056          NaN   
3       EUREKA, CA  United States        NaN         NaN          NaN   
4       Canton, OH  United States        NaN         NaN          NaN   

  Airport.Name Injury.Severity Aircraft.damage Aircraft.Category  \
0          NaN        Fatal(2)       Destroyed               NaN   
1          NaN        Fatal(4)       Destroyed               NaN   
2          NaN        Fatal(3)       Destroyed               NaN   
3          NaN        Fatal(2)       Destroyed               NaN   
4          NaN        Fatal(1)       Destroyed               NaN   

  Registration.Number      Make     Model Amateur.Built Number.of.Engines  \
0              NC6404   Stinson     108-3            No                 1   
1              N5069P     Piper  PA24-180            No                 1   
2              N5142R    Cessna      172M            No                 1   
3              N1168J  Rockwell       112            No                 1   
4               N15NY    Cessna       501            No               NaN   

     Engine.Type FAR.Description Schedule Purpose.of.flight Air.carrier  \
0  Reciprocating             NaN      NaN          Personal         NaN   
1  Reciprocating             NaN      NaN          Personal         NaN   
2  Reciprocating             NaN      NaN          Personal         NaN   
3  Reciprocating             NaN      NaN          Personal         NaN   
4            NaN             NaN      NaN          Personal         NaN   

  Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                    2                      0                    0   
1                    4                      0                    0   
2                    3                    NaN                  NaN   
3                    2                      0                    0   
4                    1                      2                  NaN   

  Total.Uninjured Weather.Condition Broad.phase.of.flight   Report.Status  \
0               0               UNK                Cruise  Probable Cause   
1               0               UNK               Unknown  Probable Cause   
2             NaN               IMC                Cruise  Probable Cause   
3               0               IMC                Cruise  Probable Cause   
4               0               VMC              Approach  Probable Cause   

  Publication.Date  
0              NaN  
1       19-09-1996  
2       26-02-2007  
3       12-09-2000  
4       16-04-1980

In [7]:
aviation_data_df.tail()

Event.Id Investigation.Type Accident.Number  Event.Date  \
88884  20221227106491           Accident      ERA23LA093  2022-12-26   
88885  20221227106494           Accident      ERA23LA095  2022-12-26   
88886  20221227106497           Accident      WPR23LA075  2022-12-26   
88887  20221227106498           Accident      WPR23LA076  2022-12-26   
88888  20221230106513           Accident      ERA23LA097  2022-12-29   

            Location        Country Latitude Longitude Airport.Code  \
88884  Annapolis, MD  United States      NaN       NaN          NaN   
88885    Hampton, NH  United States      NaN       NaN          NaN   
88886     Payson, AZ  United States  341525N  1112021W          PAN   
88887     Morgan, UT  United States      NaN       NaN          NaN   
88888     Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name Injury.Severity Aircraft.damage Aircraft.Category  \
88884          NaN           Minor             NaN               NaN   
88885          NaN             NaN             NaN               NaN   
88886       PAYSON       Non-Fatal     Substantial          Airplane   
88887          NaN             NaN             NaN               NaN   
88888          NaN           Minor             NaN               NaN   

      Registration.Number                        Make      Model  \
88884              N1867H                       PIPER  PA-28-151   
88885              N2895Z                    BELLANCA       7ECA   
88886              N749PJ  AMERICAN CHAMPION AIRCRAFT      8GCBC   
88887              N210CU                      CESSNA       210N   
88888              N9026P                       PIPER  PA-24-260   

      Amateur.Built Number.of.Engines Engine.Type FAR.Description Schedule  \
88884            No               NaN         NaN             091      NaN   
88885            No               NaN         NaN             NaN      NaN   
88886            No                 1         NaN             091      NaN   
88887            No               NaN         NaN             091      NaN   
88888            No               NaN         NaN             091      NaN   

      Purpose.of.flight         Air.carrier Total.Fatal.Injuries  \
88884          Personal                 NaN                    0   
88885               NaN                 NaN                    0   
88886          Personal                 NaN                    0   
88887          Personal  MC CESSNA 210N LLC                    0   
88888          Personal                 NaN                    0   

      Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
88884                      1                    0               0   
88885                      0                    0               0   
88886                      0                    0               1   
88887                      0                    0               0   
88888                      1                    0               1   

      Weather.Condition Broad.phase.of.flight Report.Status Publication.Date  
88884               NaN                   NaN           NaN       29-12-2022  
88885               NaN                   NaN           NaN              NaN  
88886               VMC                   NaN           NaN       27-12-2022  
88887               NaN                   NaN           NaN              NaN  
88888               NaN                   NaN           NaN       30-12-2022

In [8]:
aviation_data_df.sample(10)

Event.Id Investigation.Type Accident.Number  Event.Date  \
35609  20001206X01813           Accident      MIA94LA182  1994-07-24   
39932  20001208X05995           Accident      FTW96LA278  1996-06-28   
23974  20001213X29317           Incident      CHI89IA179  1989-09-05   
35478  20001206X01736           Incident      FTW94IA227  1994-07-08   
51429  20011010X02061           Accident      IAD01LA104  2001-09-02   
88587  20221003106050           Accident      WPR22LA367  2022-09-27   
80871  20171116X91006           Accident      ANC18LA009  2017-11-16   
25388  20001212X23080           Accident      BFO90LA041  1990-05-01   
16554  20001213X35404           Accident      MIA87LA059  1986-12-26   
13345  20001214X37972           Accident      CHI86FA017  1985-10-27   

             Location         Country   Latitude   Longitude Airport.Code  \
35609  ATLANTIC OCEAN  ATLANTIC OCEAN        NaN         NaN          PBI   
39932        KRUM, TX   United States        NaN         NaN         TA48   
23974        KNOX, IN   United States        NaN         NaN         NONE   
35478      DENVER, CO   United States        NaN         NaN          DEN   
51429     AUGUSTA, ME   United States  44.322223  -69.785555          AUG   
88587   Hermiston, OR   United States    454941N    1191533W          HRI   
80871   Opa Locka, FL   United States    255427N    0801642W          OPF   
25388   BROOKNEAL, VA   United States        NaN         NaN          OV4   
16554    KEY WEST, FL   United States        NaN         NaN          EYW   
13345  PLAINFIELD, WI   United States        NaN         NaN          NaN   

                    Airport.Name Injury.Severity Aircraft.damage  \
35609   PALM BEACH INTERNATIONAL       Non-Fatal       Destroyed   
39932             BIRD DOG FIELD       Non-Fatal     Substantial   
23974                        NaN        Incident             NaN   
35478    STAPLETON INTERNATIONAL        Incident           Minor   
51429      Augusta State Airport       Non-Fatal     Substantial   
88587             HERMISTON MUNI       Non-Fatal     Substantial   
80871  MIAMI-OPA LOCKA EXECUTIVE       Non-Fatal     Substantial   
25388  BROOKNEAL/CAMPBELL COUNTY       Non-Fatal     Substantial   
16554     KEY WEST INTERNATIONAL       Non-Fatal     Substantial   
13345                        NaN        Fatal(1)       Destroyed   

      Aircraft.Category Registration.Number                 Make  \
35609               NaN              N63317  Smith, Ted Aerostar   
39932               NaN              N47621                 Ryan   
23974               NaN              N608TW               Boeing   
35478               NaN              N83873    Mcdonnell Douglas   
51429               NaN              N65057               Cessna   
88587          Airplane              N2452F               CESSNA   
80871          Airplane              N6102R               CESSNA   
25388               NaN              N4034Z               Hiller   
16554               NaN              N2197X                Piper   
13345               NaN              N7840S                 Bell   

              Model Amateur.Built Number.of.Engines    Engine.Type  \
35609  AEROSTAR 601            No                 2  Reciprocating   
39932         PT-22            No                 1  Reciprocating   
23974       767-231            No                 2      Turbo Fan   
35478          MD80            No                 2      Turbo Jet   
51429          172P            No                 1  Reciprocating   
88587          180H            No                 1            NaN   
80871         172RG            No                 1  Reciprocating   
25388       UH-12ET            No                 1    Turbo Shaft   
16554         PA-44            No                 2  Reciprocating   
13345         47G-5            No                 1  Reciprocating   

      FAR.Description Schedule Purpose.of.flight                Air.carrier  \
35609             NaN   

In [9]:
aviation_data_df.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

In [10]:
aviation_data_df.shape

(88889, 31)

In [11]:
len(aviation_data_df.columns)

31

In [12]:
usstate_codes_df.head()

US_State Abbreviation
0     Alabama           AL
1      Alaska           AK
2     Arizona           AZ
3    Arkansas           AR
4  California           CA

In [13]:
usstate_codes_df.tail()

US_State Abbreviation
57  Virgin Islands           VI
58   Washington_DC           DC
59  Gulf of mexico           GM
60  Atlantic ocean           AO
61   Pacific ocean           PO

In [14]:
usstate_codes_df.shape

(62, 2)

In [15]:
usstate_codes_df.columns

Index(['US_State', 'Abbreviation'], dtype='object')

# DATA UNDERSTANDING



The data folder contains two data sets

## Aviation dataset
This dataset contains data of reported aviation accidents and information regarding the accident reported.

The dataset contains 31 columns with each column containing information on the accident reported.

Some of the columns in the dataset include:date of the accident, the severity of the accident, country where the accident occurred, aircraft damage, aircraft type and model etc

The dataset contains 88889 records

## US State Codes dataset
This dataset contains information on the full name of abbreviated us state codes

The dataset contains two columns i.e the US state and its abbrevaiation

The dataset contains 62 records





# Data cleaning

## Aviation Dataset

In [16]:
aviation_data_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Event.Id                88889 non-null  object
 1   Investigation.Type      88889 non-null  object
 2   Accident.Number         88889 non-null  object
 3   Event.Date              88889 non-null  object
 4   Location                88837 non-null  object
 5   Country                 88663 non-null  object
 6   Latitude                34382 non-null  object
 7   Longitude               34373 non-null  object
 8   Airport.Code            50249 non-null  object
 9   Airport.Name            52790 non-null  object
 10  Injury.Severity         87889 non-null  object
 11  Aircraft.damage         85695 non-null  object
 12  Aircraft.Category       32287 non-null  object
 13  Registration.Number     87572 non-null  object
 14  Make                    88826 non-null  object
 15  Mo

Using pd.info we can observe the following

a) every column contain string values

b)They are columns with null values

c)Columns names contain different structures and need to be standardized

### Converting column names to a simple standardized form

a)column names to be converted to lower case

b)column names to be seperated by underscore


In [18]:
#converting the columns into lower case
aviation_data_df.columns=aviation_data_df.columns.str.lower()

In [23]:
#seperating column names with an underscore rather than with a period

aviation_data_df.columns=aviation_data_df.columns.str.replace('.','_')

aviation_data_df.columns

Index(['event_id', 'investigation_type', 'accident_number', 'event_date',
       'location', 'country', 'latitude', 'longitude', 'airport_code',
       'airport_name', 'injury_severity', 'aircraft_damage',
       'aircraft_category', 'registration_number', 'make', 'model',
       'amateur_built', 'number_of_engines', 'engine_type', 'far_description',
       'schedule', 'purpose_of_flight', 'air_carrier', 'total_fatal_injuries',
       'total_serious_injuries', 'total_minor_injuries', 'total_uninjured',
       'weather_condition', 'broad_phase_of_flight', 'report_status',
       'publication_date'],
      dtype='object')

In [24]:
aviation_data_df.head()

event_id investigation_type accident_number  event_date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          location        country   latitude   longitude airport_code  \
0  MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2    Saltville, VA  United States  36.922223  -81.878056          NaN   
3       EUREKA, CA  United States        NaN         NaN          NaN   
4       Canton, OH  United States        NaN         NaN          NaN   

  airport_name injury_severity aircraft_damage aircraft_category  \
0          NaN        Fatal(2)       Destroyed               NaN   
1          NaN        Fatal(4)       Destroyed               NaN   
2          NaN        Fatal(3)       Destroyed               NaN   
3          NaN        Fatal(2)       Destroyed               NaN   
4          NaN        Fatal(1)       Destroyed               NaN   

  registration_number      make     model amateur_built number_of_engines  \
0              NC6404   Stinson     108-3            No                 1   
1              N5069P     Piper  PA24-180            No                 1   
2              N5142R    Cessna      172M            No                 1   
3              N1168J  Rockwell       112            No                 1   
4               N15NY    Cessna       501            No               NaN   

     engine_type far_description schedule purpose_of_flight air_carrier  \
0  Reciprocating             NaN      NaN          Personal         NaN   
1  Reciprocating             NaN      NaN          Personal         NaN   
2  Reciprocating             NaN      NaN          Personal         NaN   
3  Reciprocating             NaN      NaN          Personal         NaN   
4            NaN             NaN      NaN          Personal         NaN   

  total_fatal_injuries total_serious_injuries total_minor_injuries  \
0                    2                      0                    0   
1                    4                      0                    0   
2                    3                    NaN                  NaN   
3                    2                      0                    0   
4                    1                      2                  NaN   

  total_uninjured weather_condition broad_phase_of_flight   report_status  \
0               0               UNK                Cruise  Probable Cause   
1               0               UNK               Unknown  Probable Cause   
2             NaN               IMC                Cruise  Probable Cause   
3               0               IMC                Cruise  Probable Cause   
4               0               VMC              Approach  Probable Cause   

  publication_date  
0              NaN  
1       19-09-1996  
2       26-02-2007  
3       12-09-2000  
4       16-04-1980

### All columns contain strings values

This columns were all converted to string values while reading the csv so to clear error generated.Therefore we need to go through each column and with intuition convert some of the columns back to their original data type

In [26]:
aviation_data_df.sample(10)

event_id investigation_type accident_number  event_date  \
20989  20001213X26153           Accident      CHI88DEV03  1988-07-28   
54795  20030610X00841           Accident      ANC03LA056  2003-06-08   
15946  20001213X34719           Accident      ATL86LA262  1986-09-20   
46481  20001212X19293           Accident      LAX99LA240  1999-07-01   
66590  20090622X61140           Accident      WPR09CA303  2009-06-20   
15896  20001213X34865           Accident      NYC86LA236  1986-09-14   
36507  20001206X02672           Accident      ANC95LA018  1994-12-12   
1208   20020917X02920           Accident      CHI82DA170  1982-05-15   
32543  20001211X12112           Accident      DEN93FA043  1993-04-16   
63802  20071227X02000           Accident      NYC08FA051  2007-12-04   

              location        country   latitude    longitude airport_code  \
20989      ASHLAND, IL  United States        NaN          NaN          NaN   
54795       TELLER, AK  United States  65.406111  -164.655556          5QC   
15946    TULLAHOMA, TN  United States        NaN          NaN          THA   
46481      POLACCA, AZ  United States        NaN          NaN          4PH   
66590        Ogden, UT  United States    411146N     0112044W          OGD   
15896  PEDRICKTOWN, NJ  United States        NaN          NaN          NaN   
36507       PALMER, AK  United States        NaN          NaN          NaN   
1208     PENDLETON, IN  United States        NaN          NaN         IN32   
32543       ELBERT, CO  United States        NaN          NaN          NaN   
63802   New Castle, DE  United States  39.675556   -75.612222          ILG   

             airport_name injury_severity aircraft_damage aircraft_category  \
20989                 NaN       Non-Fatal     Substantial               NaN   
54795        Quartz Creek       Non-Fatal     Substantial               NaN   
15946           TULLAHOMA       Non-Fatal     Substantial               NaN   
46481    POLACCA AIRSTRIP       Non-Fatal     Substantial               NaN   
66590      Ogden-Hinckley       Non-Fatal     Substantial          Airplane   
15896     OLD MAN AIRPORT       Non-Fatal     Substantial               NaN   
36507                 NaN       Non-Fatal     Substantial               NaN   
1208           STEPHENSON       Non-Fatal     Substantial          Airplane   
32543                 NaN        Fatal(2)       Destroyed               NaN   
63802  New Castle Airport        Fatal(1)       Destroyed          Airplane   

      registration_number       make      model amateur_built  \
20989              N9033F     Hughes      369HS            No   
54795              N2475M      Piper      PA-12            No   
15946               N6877  Alexander     AS-K13            No   
46481              N5943M     Cessna       421B            No   
66590              N731GE     CESSNA      A188B            No   
15896              N738HZ     Cessna       172N            No   
36507              N717DR      Piper      PA-18            No   
1208                N4652   Playmate      SA11A           Yes   
32543              N38481      Piper  PA-28-181            No   
63802              N105PP      Beech         60            No   

      number_of_engines    engine_type            far_description schedule  \
20989                 1    Turbo Shaft                        NaN      UNK   
54795                 1  Reciprocating                        NaN      NaN   
15946                 0        Unknown                        NaN      NaN   
46481                 2  Reciprocating                        NaN      NaN   
66590                 1  Reciprocating                        137      NaN   
15896                 1  Reciprocating                        NaN      NaN   
36507                 1  Reciprocating                        NaN      NaN   
1208                  1  Reciprocating  Part 91: General Aviation      NaN   
32543                 1  Reciprocating                        NaN      Na

Taking a sample of the dataset we can safely deduce that the following columns originally contained interger values

number_of_engines,total_fatal_injuries,total_minor_injuries,total_serious_injuries and total_uninjured

In [35]:
#converting the above columns into interger

aviation_data_df = aviation_data_df.astype({'number_of_engines':'int','total_fatal_injuries':'int','total_serious_injuries':'int','total_minor_injuries':'int','total_uninjured':'int'})

ValueError: cannot convert float NaN to integer